error: invalid use of non-static data member 'u'

I wrote a custom cmake for this project, but it errors on my Clang 3.4 compiler on OSX Mavericks.

```
eixx/src/../include/eixx/marshal/ref.hpp:199:84: error: invalid use of non-static data member 'u'
               ::memcmp(&m_blob->data()->u, &t.m_blob->data()->u, sizeof(ref_blob::u)) == 0;
                                                                         ~~~~~~~~~~^
eixx/src/../include/eixx/marshal/eterm.ipp:79:37: note: in instantiation of member function 'eixx::marshal::ref<std::__1::allocator<char> >::operator==' requested here
        case REF:    return vt.r    == rhs.vt.r;
                                    ^
eixx/src/../include/eixx/marshal/visit_match.hpp:65:23: note: in instantiation of member function 'eixx::marshal::eterm<std::__1::allocator<char> >::operator==' requested here
            return et == m_pattern;
                      ^
eixx/src/../include/eixx/marshal/visit.hpp:50:23: note: in instantiation of function template specialization 'eixx::marshal::visit_eterm_match<std::__1::allocator<char> >::operator()<long>' requested here
            : result( v(t1) )
                      ^
eixx/src/../include/eixx/marshal/eterm.hpp:570:33: note: in instantiation of function template specialization 'eixx::marshal::wrap<bool, eixx::marshal::visit_eterm_match<std::__1::allocator<char> > >::wrap<long>' requested here
            case LONG:   return wrapper(v, vt.i);
                                ^
eixx/src/../include/eixx/marshal/visit.hpp:66:19: note: in instantiation of function template specialization 'eixx::marshal::eterm<std::__1::allocator<char> >::visit<eixx::marshal::visit_eterm_match<std::__1::allocator<char> > >' requested here
        return et.visit(*static_cast<const Derived*>(this)).result;
                  ^
eixx/src/../include/eixx/marshal/eterm.ipp:263:22: note: in instantiation of function template specialization 'eixx::marshal::static_visitor<eixx::marshal::visit_eterm_match<std::__1::allocator<char> >, bool>::apply_visitor<std::__1::allocator<char> >' requested here
        if (!visitor.apply_visitor(*this))
                     ^
eixx/src/test_node.cpp:43:21: note: in instantiation of member function 'eixx::marshal::eterm<std::__1::allocator<char> >::match' requested here
    if (s_put_chars.match(a_msg->msg(), &l_binding))
                    ^
6 warnings and 2 errors generated.
make[2]: *** [src/CMakeFiles/test_node.dir/test_node.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/test_node.dir/all] Error 2
make: *** [all] Error 2
16:34:33: The process "/usr/bin/make" exited with code 2.
Error while building/deploying project eixx (kit: Desktop Qt 5.2.1 clang 64bit)
When executing step "Make"
16:34:33: Elapsed time: 00:10.
```

--- a/include/eixx/marshal/ref.hpp
+++ b/include/eixx/marshal/ref.hpp
@@ -200,7 +200,7 @@ public:
 
     bool operator==(const ref<Alloc>& t) const {
         return node() == t.node() &&
-               ::memcmp(&m_blob->data()->u, &t.m_blob->data()->u, sizeof(ref_blob::u)) == 0;
+               ::memcmp(&m_blob->data()->u, &t.m_blob->data()->u, sizeof(&m_blob->data()->u)) == 0;
     }
 
     /// Less operator, needed for maps

